prescient.el

```diff
@@ -714,8 +717,277 @@ Return the sorted list. The original is modified destructively."
              prescient-aggressive-file-save)
     (prescient--save)))
 
-;;;; Closing remarks
+;;;; Completion Style
+
+;; This section contains functions and user options for implementing
+;; the prescient.el completion style.  This section based on
+;; Orderless.el.  See: https://github.com/oantolin/orderless
+
+;;;;; User options
+
+(defcustom prescient-highlight-completions t
+  "Whether `prescient.el' should highlight completion candidates.
+
+This affects the completion style, not other methods."
+  :type 'boolean)
+
+(defcustom prescient-completion-enable-sort t
+  "Whether the prescient.el completion style can use `prescient-sort'.
+
+If non-nil, then completion tables that don't specify a sorting
+function will use prescient.el for sorting.  Table that do
+specify a sorting function are unaffected.
+
+This option must be non-nil to use
+`prescient-sort-full-matches-first' with the completion style,
+since that option depends on the input used for filtering.
+
+Some completion UIs provide their own user options for sorting
+functions.  If you only wish to sort by frequency and recency via
+the function `prescient-sort' (which does not depend on filtering
+input), then you might not need this option.
 
+See also the user option `prescient-sort-length-enable'."
+  :type 'boolean)
+
+(defface prescient-primary-highlight
+  '((t :weight bold))
+  "Face used to highlight the parts of candidates that match the input.")
+
+(defface prescient-secondary-highlight
+  '((t :inherit prescient-primary-highlight :underline t))
+  "Additional face used to highlight parts of candidates.
+
+May be used to highlight parts of candidates that match specific
+parts of the input.")
+
+;;;;; Utility functions
+
+;; From Orderless
+;; TODO: Not sure what this is for.
+(defun prescient--anchored-quoted-regexp (regexp)
+  "Determine if REGEXP is a quoted regexp anchored at the beginning.
```

Comment:
See https://github.com/oantolin/orderless/commit/793eb0ec01ece8ba2a8073411331b03d13db6d47 and the associated discussion.

Reply:
@minad, would you please post an example of the Orderless style dispatcher that is mentioned in these discussions? I did not see it mentioned in the README. Are people only pointed to it when `completion-styles` as `(orderless basic)` is insufficient?

Reply:
Do you have the concept of style dispatchers in prescient? If not, I suggest removing this part of the code. In orderless the idea is to map the first "word" to "^word" via the dispatcher. Then regexp "^word" is mapped back to "word" and passed as prefix to the completion table.